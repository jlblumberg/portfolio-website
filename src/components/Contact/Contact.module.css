.flexContainer { 
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
  background-color: rgb(245, 245, 245);
}

.form {
  width: 50%;
}