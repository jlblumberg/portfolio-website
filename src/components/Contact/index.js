import React from 'react';
import styles from './Contact.module.css'
import { IconButton, Typography, TextField, Button } from '@material-ui/core';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';

const Contact = () => {

  const handleClick = () => {
    alert('add in scroll functionality later')
  }

  return (
    <div className={styles.flexContainer}>
      <div id='header' className={styles.headerText}>
        <Typography style={{ marginTop: '5.5em', marginBottom: '3.5em' }} variant='h5'>
          Why not say <strong>hi</strong> ðŸ‘‹
        </Typography>
      </div>
      <div className={styles.form} id='form'>
        <form action="https://getsimpleform.com/messages?form_api_token=fee2c6c94fa2bb6ca482405d707eba90" method='POST'>
          <TextField fullWidth style={{ marginBottom: '1.25em' }} variant='outlined' name='email' label='Your email address' />
          <TextField fullWidth style={{ marginBottom: '1.25em' }} variant='outlined' name='subject' label='Message summary' />
          <TextField fullWidth style={{ marginBottom: '2em' }} variant='outlined' multiline rows={4} name='body' label='Message body' />
          <div style={{ textAlign: 'center' }}>
            <Button
              color='primary'
              variant='contained'
              type='submit'
              onClick={(e) => { alert("Thanks! I'll get back to you as soon as I can.") }}
            >
              send
            </Button>
          </div>
        </form>
      </div>
      <div className={styles.upButton}>
        <IconButton
          color='primary'
          id='back-to-top-button'
          onClick={handleClick}
        >
          <KeyboardArrowUpIcon fontSize='large' />
        </IconButton>
      </div>
    </div>
  )
}

export default Contact