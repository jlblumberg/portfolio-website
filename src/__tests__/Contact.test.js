import React from "react";
import { shallow } from "enzyme";
import Contact from "../components/Contact";

describe("Contact", () => {
  const wrapper = shallow(<Contact />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a header and a form", () => {
    expect(wrapper.find("#header").length).toEqual(1);
  });

  it("has an up button which takes you to the top", () => {
    expect(wrapper.find("#back-to-top-button").exists()).toBe(true);
  });
});
