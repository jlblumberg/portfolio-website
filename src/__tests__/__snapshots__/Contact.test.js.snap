// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `ShallowWrapper {}`;
